component_definitions.go

```diff
@@ -336,6 +350,8 @@ func (nds *ComponentDefinitions) ComponentNames() ComponentNames {
 //   - group component definitions that share same pod
 //   - prevent duplicated lists, once a component definition is present in one
 //     list, it is not present in other lists.
+// The resulting maps are sorted such that components that link to other components are
+// found after the component that they link to.
 func (nds *ComponentDefinitions) AllDefsPerPod(names ComponentNames) ([]ComponentDefinitions, error) {
```

Comment:
So this is like a groupByPod function? I know, replacing one name with another is not really helpful, but I am trying to get behind the goal here.

So, our ComponentDefinitions contains 3 components:
-  A (Pod "1")
-  B
-  C (Pod "1")

This function will return a slice with 2 items:
- A + C
- B

right?